stan/math/prim/prob/inv_wishart_lpdf.hpp

```diff
@@ -58,12 +58,14 @@ return_type_t<T_y, T_dof, T_scale> inv_wishart_lpdf(const T_y& W,
   Eigen::Index k = S.rows();
   T_nu_ref nu_ref = nu;
   T_S_ref S_ref = S;
+  T_W_ref W_ref = W;
   check_greater(function, "Degrees of freedom parameter", nu_ref, k - 1);
+  check_symmetric(function, "random variable", W_ref);
+  check_symmetric(function, "scale parameter", S_ref);
 
-  LDLT_factor<value_type_t<T_y>, Eigen::Dynamic, Eigen::Dynamic> ldlt_W(W);
+  const auto& ldlt_W = make_ldlt_factor<T_y, T_dof, T_scale>(W_ref);
```

Comment:
I'm not understanding this switch to depending on multiple types. And why is this `const auto&`? And should these be using the `T_*_ref` types?

Reply:
Got rid of the multiple type dependency